File: C++/Remove Duplicates from Sorted List.cpp
#include<iostream>

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};


ListNode *deleteDuplicates(ListNode *head)
{
    if(head==NULL||head->next==NULL) return head;
    ListNode *node1=head->next,*node2=head;
    while(node1!=NULL)
        {
            if(node1->val==node2->val)
                {
                    node2->next=node1->next;
                    node1=node1->next;
                }
            else
                {
                    node2=node1;
                    node1=node1->next;
                }
        }
    return head;
}

File: Python/Remove-Duplicates-from-Sorted-List.py
#!/usr/bin/env python

# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    # @param head, a ListNode
    # @return a ListNode
    def deleteDuplicates(self, head):
        if head == None:
            return None
        node1 = head
        node2 = head.next
        while node2 != None :
            if node2.val != node1.val:
                node1.next = node2
                node1 = node1.next
                node2 = node2.next
            else:
                node2 = node2.next

        node1.next=None
        return head